test/tests/postprocessors/current_boundary_id/current_boundary_id.i

```diff
@@ -0,0 +1,57 @@
+#
+# This is used to create the mesh but it does not work with --distributed-mesh flag
+# and the parallel test bombs.
+#
+#[Mesh]
+#  type = MeshGeneratorMesh
+#
+#  [./cartesian]
+#    type = CartesianMeshGenerator
+#    dim = 2
+#    dx = '1 1'
+#    dy = '1'
+#    subdomain_id = '1 2'
+#  [../]
+#
+#  [./interior_bc]
+#    type = SideSetsBetweenSubdomainsGenerator
+#    master_block = 1
+#    paired_block = 2
+#    new_boundary = 12
+#    input = cartesian
+#  [../]
+#[]
+
+[Mesh]
+  type = FileMesh
+  file = current_boundary_id_in.e
+[]
+
+[Problem]
+  kernel_coverage_check = false
+[]
+
+[Variables]
+  [./u]
+  [../]
+[]
+
+[Postprocessors]
+  [./interior_boundary]
+    type = AverageBID
+    boundary = 12
+  [../]
+
+  [./top_boundary]
+    type = AverageBID
+    boundary = top
+  [../]
+[]
+
+[Executioner]
+  type = Steady
+[]
+
+[Outputs]
+  csv = true
+[]
```

Comment:
When I saw `AverageBID` I was like WTF? Either explain what the test is trying to achieve or design a better one.

For example, if you want to make sure that the right boundary ID is available in side-restricted objects, you could use an AuxBC on a side that would store the boundary ID. Diff on a field variable would be much better than a global reduction. You can have one AuxBC per variable and side

Reply:
This was a good point. I tried `AuxBC` and it didn't work. Turns out, I needed to change the corresponding threaded loop class. See if this test is better.